src/main/java/persistence/entity/impl/EntityManagerImpl.java

```diff
@@ -0,0 +1,52 @@
+package persistence.entity.impl;
+
+import java.sql.Connection;
+import jdbc.TransactionalJdbcTemplate;
+import persistence.entity.EntityManger;
+import persistence.entity.PersistenceContext;
+
+public class EntityManagerImpl implements EntityManger {
+
+    private final PersistenceContext persistenceContext;
+    private final TransactionalJdbcTemplate transactionalJdbcTemplate;
```

Comment:
```suggestion
    private final JdbcTemplate transactionalJdbcTemplate;
```

`TransactionalJdbcTemplate` 타입을 명확히 해야하는 이유가 궁금해요~

Reply:
처음에는 자식 타입을 사용하는 것이 객체의 역할을 명확하게 파악하는 데 도움이 된다고 생각했습니다. 하지만 다시 생각해보니, 부모 타입을 명시하고 구현체를 유연하게 교체할 수 있는 설계가 더 바람직하다는 생각이 듭니다. 또한, 객체를 초기화하는 과정에서 어떤 구현체로 초기화되는지 명확히 알 수 있기 때문에 자식 타입보다 부모 타입을 사용하는 것이 더 나은 선택이라고 느꼈습니다.

혹시 제가 놓친 부분이나 잘못된 생각이 있다면, 피드백을 주시면 감사하겠습니다.

Reply:
TransactionalJdbcTemplate이 JdbcTemplate을 확장해서 사용하고 있어서 타입을 TransactionalJdbcTemplate로 선언하지 않으면 beginTransaction 과 같은 메서드 들은 사용할 수 없더라구요. 필요할때마다 명시적으로 cast해서 사용하는 방법은 보일러플레이트가 생길거 같아보였습니다.

혹시 제가 놓친 부분이나 잘못된 생각이 있다면, 피드백을 주시면 감사하겠습니다.

Reply:
또 한편으로는, 부모 타입을 명시하고 구현체를 유연하게 교체할 수 있는 설계도 좋아보이는 거 같습니다. 또한, 객체를 초기화하는 과정에서 어떤 구현체로 초기화되는지 명확히 알 수 있기 때문에 자식 타입보다 부모 타입을 사용하는 것이 더 나은 선택이 될수 있다고 생각이 드네요.

리뷰어님께서라면 이 상황에서 어떤 방식으로 구현하실지 궁금합니다! :)